src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp

```diff
@@ -863,34 +867,40 @@ bool ControllerScriptEngineLegacy::evaluateScriptFile(const QFileInfo& scriptFil
 }
 
 #ifdef MIXXX_USE_QML
-std::shared_ptr<QQuickItem> ControllerScriptEngineLegacy::loadQMLFile(
+bool ControllerScriptEngineLegacy::instanciateQMLComponent(
```

Comment:
So this function here is a bit of a mess right now but I don't see how simplify this, unless I make screen delcaration a part of `MixxxController`, which is doable I think?

Reply:
I'm not sure moving the scene binding logic in this function helps much to be honest. Perhaps you should move the logic from line `460` to a new function, but I would suggest not making  `loadQMLFile` more complex.